kafka-rest/src/main/java/io/confluent/kafkarest/backends/kafka/KafkaModule.java

```diff
@@ -130,11 +130,17 @@ public void dispose(Producer<byte[], byte[]> producer) {
 
   private static final class ProducerMetricsFactory implements Factory<ProducerMetrics> {
 
+    private final Provider<KafkaRestContext> context;
     private ProducerMetrics producerMetrics;
 
+    @Inject
+    ProducerMetricsFactory(Provider<KafkaRestContext> context) {
+      this.context = requireNonNull(context);
+    }
+
     @Override
     public ProducerMetrics provide() {
-      producerMetrics = new ProducerMetrics(Time.SYSTEM);
+      producerMetrics = new ProducerMetrics(context.get().getConfig(), Time.SYSTEM);
       return producerMetrics;
     }
```

Comment:
Did you find a nicer approach than what we discussed offline about initializing `producerMetrics` and ensuring that's done only once, and done in a thread-safe manner?
If not I'd suggest using a DCL-like structure that can ensure we are not reinitializing the metrics multiple times.
In its simplest form (when relying on the singleton factory-singleton provided object) this will just check that `producerMetrics` is always `null` when we try to initialize it.

Reply:
Hm, I think I got an outdated diff in the mix, sorry about that. Let me check all my other comments too.